packages/data-provider/src/services/categoryService.ts

```diff
@@ -0,0 +1,43 @@
+import * as _ from 'lodash-es';
+import { BaseDependencies } from '../config/config.js';
+import categoryRepository from '../repository/categoryRepository.js';
+import submittedDataRepository from '../repository/submittedRepository.js';
+import { CategoryDetailsResponse, ListAllCategoriesResponse } from '../utils/types.js';
+
+const categoryService = (dependencies: BaseDependencies) => {
+	const LOG_MODULE = 'CATEGORY_SERVICE';
+	const { logger } = dependencies;
+	const categoryRepo = categoryRepository(dependencies);
+	const submittedDataRepo = submittedDataRepository(dependencies);
+	return {
+		getDetails: async (categoryId: number): Promise<CategoryDetailsResponse | undefined> => {
+			logger.debug(LOG_MODULE, `Get category Details`);
+			const category = await categoryRepo.getCategoryById(categoryId);
+			if (category) {
+				const organizationsFound = await submittedDataRepo.getAllOrganizationsByCategoryId(categoryId);
+
+				return {
+					id: category.id,
+					dictionary: {
+						name: category.activeDictionary?.name || '',
+						version: category.activeDictionary?.version || '',
+					},
+					name: category.name,
+					organizations: organizationsFound,
+					createdAt: _.toString(category.createdAt?.toISOString()),
+					createdBy: _.toString(category.createdBy),
+					updatedAt: _.toString(category.updatedAt),
+					updatedBy: _.toString(category.updatedBy),
+				};
+			}
```

Comment:
What is the reason for converting so many of these values to different types form how they are actually stored? I do like that we have a different `Category` type from `CategoryDetailsResponse`, and I understand converting dates to strings,  but it seems weird that we are converting some values from `undefined` to `''`. This is happening within the `dictionary` property, so I assume this is a quirk of the situation where a category could fail to join with an `activeDictionary`? When `category.activeDictionary` is undefined, perhaps just return `dictionary: undefined` instead?

Reply:
The only reason why this conversion of `name` and `version` to `''` is when a join between the category and activeDictionary fails, as you mentioned in your comment and it would make more sense to change dictionary as optional as suggested.